pyqtorch/modules/circuit.py

```diff
@@ -40,6 +40,22 @@ def __init__(self, n_qubits: int, operations: list):
         self.n_qubits = n_qubits
         self.operations = torch.nn.ModuleList(operations)
 
+    def __mul__(self, other: QuantumCircuit) -> QuantumCircuit:
+        if self.n_qubits != other.n_qubits:
+            raise ValueError(
+                (
+                    f"Number of Qubits don't match. "
+                    f"Left gate is applied on a {self.n_qubits} qubit system and "
+                    f"right gate is applied on a {other.n_qubits} qubit system."
+                )
+            )
+
+        if isinstance(other, QuantumCircuit):
+            return QuantumCircuit(self.n_qubits, other.operations.append(self.operations))
+
+        else:
+            return NotImplemented
```

Comment:
It would be great to also be able to do `circuit * gate`

Reply:
Right. The way I had thought about it was a gate being applied on a circuit (aka after every other gate). But it makes sense to allow a gate to be inserted anywhere, in this case at the beginning. Is this what you meant?